java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanTask.java

```diff
@@ -29,14 +29,7 @@
 public interface ScanTask extends AutoCloseable {
```

Comment:
N.B. we should eventually remove this class entirely as the corresponding interface no longer exists in C++ ARROW-15745

Reply:
I can assign to myself. Thanks.